### ISSUE TITLE
Factorplot hue col/row combination results in ValueError: Collections can only map rank 1 arrays

### ISSUE THREAD

**Issue by sinkpoint** [mwaskom/seaborn#709]
Hi I'm experiencing this error when upgrading to seaborn 0.6.0. 
Removing the hue parameter makes it ok. 

The data works fine in 0.5.0

sample data is here:
https://www.dropbox.com/s/lekfpee922t4wmo/_allwchg.csv?dl=0

**Comment by mwaskom** [comment]
Thanks for posting sample data but I have no idea what you are trying to do when you get the error...

**Comment by sinkpoint** [comment]
Ops, sometime like this:
```
g = sn.factorplot('transform-type', 'metric',  data=mydata, hue='deformed', col='transform', row='metric-type', sharey=False)
```
Thanks

**Comment by cviner** [comment]
This is actually the same error that I get when I attempt to create a `factorplot`, but am able to create it using the same data via directly using `FacetGrid` and `pointplot`. I have not looked into this in detail and have yet to determine the likely cause of this error.

**Comment by cviner** [comment]
It appears that this may have to do with `factorplot`'s legend creation:

```
Traceback (most recent call last):
  File "./createFacetedLinePlots.py", line 406, in <module>
    palette=colour_pal, ci=None, linewidth=LINE_WIDTH)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/seaborn/categorical.py", line 2725, in factorplot
    g.add_legend(title=hue, label_order=hue_order)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/seaborn/axisgrid.py", line 92, in add_legend
    plt.draw()
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/pyplot.py", line 571, in draw
    get_current_fig_manager().canvas.draw()
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 469, in draw
    self.figure.draw(self.renderer)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/figure.py", line 1085, in draw
    func(*args)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2110, in draw
    a.draw(renderer)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/collections.py", line 772, in draw
    Collection.draw(self, renderer)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/collections.py", line 259, in draw
    self.update_scalarmappable()
  File "/mnt/work1/users/home2/cviner/.local/lib/python2.7/site-packages/matplotlib/collections.py", line 672, in update_scalarmappable
    raise ValueError('Collections can only map rank 1 arrays')
ValueError: Collections can only map rank 1 arrays
```

**Comment by mwaskom** [comment]
I have seen this a couple of times, but I've never been able to figure it out. It's confusing because it seems to happen at draw time, making it difficult to use the debugger.

**Comment by cviner** [comment]
I thought it might have had to do with my matplotlib version being outdated (1.2.1), but I have reproduced this error with matplotlib 1.4.3.

This is currently precluding my use of `factorplot`, so if you have any ideas for things I should test, I'd be happy to look into it.

The only hints from searching for this error have largely concerned the use of a [`matrix`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.matrix.html) instead of a (necessary) [`ndarray`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.html), but I'm not sure if that could apply in this case (although I suspect not).

**Comment by olgabot** [comment]
This is happening to me too. And it's **only** with `sns.factorplot(... kind='point')`, not `bar`. Here's a [full example](http://nbviewer.ipython.org/github/olgabot/seaborn_factorplot_pointplot_valueerror/blob/master/factorplot_valueerror.ipynb). There's also screwy behavior if you `g.map(sns.pointplot, ...)` where it doesn't plot the entire x-axis of values

**Comment by mwaskom** [comment]
I can't replicate this issue on matplotlib 1.5 using your data and code @olgabot. It's possible this was a matplotlib bug that has been fixed?

**Comment by joelostblom** [comment]
I was getting this error message recently with seaborn 0.6 and matplotlib 1.4.3 when using `pointplot` without the `hue` parameter. I downgraded to 0.5.1 and `pointplot` started working. I then upgraded to seaborn 0.6 again via `conda`, which automatically updated my matplotlib to 1.5.0 (and numpy and pandas) and `pointplot` is now working. So it might have been an issue with matplotlib 1.4.3 as @mwaskom mentioned.
